src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -726,13 +726,32 @@ static int get_pool_offset(sd_dhcp_server *server, be32_t requested_ip) {
         return be32toh(requested_ip & ~server->netmask) - server->pool_offset;
 }
 
+static int prepare_new_lease(DHCPLease **lease, be32_t address, DHCPClientId *client_id, uint8_t *chaddr, be32_t gateway, usec_t expiration){
+        *lease = new0(DHCPLease, 1);
```

Comment:
according to our coding style our functions should either succeed and then fill in return parameters. Or they should fail but then leave the return parameters unclobbered. See: https://systemd.io/CODING_STYLE – the section about "Code Organzation and Semantics".

hence, please allocate the DHCPLease object into a local variable first, and then only pass ownership to `*lease` when fully successful.